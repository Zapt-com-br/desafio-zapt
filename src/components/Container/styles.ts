import styled from 'styled-components';
import * as ButtonStyles from 'components/Button/styles';

export const Wrapper = styled.section`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  position: relative;
`;

export const ButtonWrapper = styled.div`
  position: absolute;
  bottom: 4rem;

  ${ButtonStyles.Wrapper} + ${ButtonStyles.Wrapper} {
    margin-left: 0.8rem;
  }
`;
