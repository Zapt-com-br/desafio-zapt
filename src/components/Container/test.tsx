import { render } from 'utils/test-utils';

import theme from 'styles/theme';

import Container from '.';

describe('<Container />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Container>
        <span>Desafio Zapt</span>
      </Container>
    );

    expect(container.firstChild).toHaveStyleRule(
      'max-width',
      theme.grid.container
    );

    expect(container.firstChild).toMatchInlineSnapshot(`
      .c0 {
        width: 100%;
        max-width: 120rem;
        margin-left: auto;
        margin-right: auto;
        padding-left: calc(3.2rem / 2);
        padding-right: calc(3.2rem / 2);
      }

      <div
        class="c0"
      >
        <span>
          Desafio Zapt
        </span>
      </div>
    `);
  });
});
