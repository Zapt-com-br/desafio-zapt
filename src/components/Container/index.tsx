import React, { Fragment } from 'react';
import { useHistory } from 'react-router-dom';

import { Button } from 'components';

import * as Styled from './styles';

export type ContainerProps = {
  children: React.ReactNode | React.ReactNodeArray;
  nextComponentUrl?: string;
  prevComponentUrl?: string;
};

const Container = ({
  nextComponentUrl = '',
  children,
  prevComponentUrl = '',
}: ContainerProps) => {
  const history = useHistory();
  return (
    <Styled.Wrapper>
      {children}
      <Fragment>
        {location.pathname === '/usercomments' && (
          <Styled.ButtonWrapper>
            <Button
              label="Componente anterior"
              onClick={() => history.push(prevComponentUrl)}
            />
          </Styled.ButtonWrapper>
        )}

        {location.pathname === '/' && (
          <Styled.ButtonWrapper>
            <Button
              label="Proximo componente"
              onClick={() => history.push(nextComponentUrl)}
            />
          </Styled.ButtonWrapper>
        )}

        {location.pathname === '/socialcomments' && (
          <Styled.ButtonWrapper>
            <Button
              label="Componente anterior"
              onClick={() => history.push(prevComponentUrl)}
            />
            <Button
              label="Proximo anterior"
              onClick={() => history.push(nextComponentUrl)}
            />
          </Styled.ButtonWrapper>
        )}
      </Fragment>
    </Styled.Wrapper>
  );
};

export default Container;
