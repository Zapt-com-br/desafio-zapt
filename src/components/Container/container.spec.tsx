import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithTheme } from 'helpers/renderWithThame';

import Container from '.';

describe('<Container />', () => {
  it('should render Container', () => {
    renderWithTheme(
      <Container>
        <h1>container</h1>
      </Container>
    );

    const element = screen.getByRole('heading', { name: /container/i })
      .parentElement;

    expect(element).toBeInTheDocument();
    expect(element).toMatchSnapshot();
  });

  it('should render Container whith height 100vh', () => {
    renderWithTheme(
      <Container>
        <h1>container</h1>
      </Container>
    );

    const element = screen.getByRole('heading', { name: /container/i })
      .parentElement;

    expect(element).toHaveStyleRule('height', '100vh');
  });

  it('should render Container whith display flex and align center elements', () => {
    renderWithTheme(
      <Container>
        <h1>container</h1>
      </Container>
    );

    const element = screen.getByRole('heading', { name: /container/i })
      .parentElement;

    expect(element).toHaveStyle({
      display: 'flex',
      'align-items': 'center',
      'justify-content': 'center',
    });
  });
});
