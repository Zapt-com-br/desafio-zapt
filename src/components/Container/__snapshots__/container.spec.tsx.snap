// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render Container 1`] = `
.c4 {
  font-weight: 400;
  color: #ffffff;
  font-size: 1.4rem;
}

.c3 {
  background: #334BC8;
  padding: 1.6rem 3.2rem;
  border: 0;
  border-radius: 0.4rem;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  position: relative;
}

.c1 {
  position: absolute;
  bottom: 4rem;
}

.c1 .c2 + .c2 {
  margin-left: 0.8rem;
}

<section
  class="c0"
>
  <h1>
    container
  </h1>
  <div
    class="c1"
  >
    <button
      class="c2 c3"
    >
      <span
        class="c4"
        color="white"
      >
        Proximo componente
      </span>
    </button>
  </div>
</section>
`;
